bundles/org.eclipse.core.runtime/src/org/eclipse/core/runtime/Plugin.java

```diff
@@ -526,6 +527,35 @@ public void setDebugging(boolean value) {
 		}
 	}
 
+	/**
+	 * Try to acquire a single service from the OSGi service factory (e.g.
+	 * DebugOptions, IWorkspace, IPreferenceService, ...)
+	 *
+	 * @param service the service to acquire
+	 * @return an optional representing the acquired service or an empty optional if
+	 *         there is (currently) no service available or the plugin is shut down
+	 * @since 3.25
+	 */
+	public <T> Optional<T> getOSGiService(Class<T> service) {
```

Comment:
"service" should be better "serviceClass", otherwise it sounds strange to ask this method for a service giving service as argument.

Reply:
done